web/src/app/tokens/tokens-list/tokens-list.component.ts

```diff
@@ -33,19 +37,25 @@ export class TokensListComponent {
   }
 
   // This function delete the project token
-  delete(token: ProjectTokenModel): void {
-    this.projectTokenService
-      .delete(this.projectUid, token.uid)
-      .pipe(
-        take(1)
-      )
-      .subscribe(() => {
-        this.projectTokenService.findAll(this.projectUid)
+  delete(tokenUid: string): void {
+    let dialogConfig: MatDialogConfig = new MatDialogConfig();
+    dialogConfig = {
+      width: '500px',
+      data: {
+        title: 'Delete Token',
+        content: 'Are you sure you want to delete?',
+      },
+    };
+    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
+    this.dialogRef.afterClosed().subscribe((result: boolean) => {
+      if (result === true) {
+        this.projectTokenService
+          .delete(this.projectUid, tokenUid)
           .pipe(
             take(1)
-          )
-          .subscribe((data: ProjectTokenModel[]) => this.tokenList = data);
-      });
+          ).
```

Comment:
`.` on wrong line. 

`take` not needed, as first is used on `delete` in service